Core/src/Navigation/SurfaceArrayNavigationPolicy.cpp

```diff
@@ -37,6 +37,13 @@ SurfaceArrayNavigationPolicy::SurfaceArrayNavigationPolicy(
     surfaces.push_back(surface.getSharedPtr());
   }
 
+  ACTS_VERBOSE("Number of surfaces passed to the surface array creation: "
+               << surfaces.size());
+  if (surfaces.size() == 0) {
```

Comment:
One last thing before I merge this:

```suggestion
  if (surfaces.empty()) {
```